File: bddtests/src/main/kotlin/mocking/tpp/models/Event.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "Event")
@XmlAccessorType(XmlAccessType.FIELD)
data class Event(
        @XmlAttribute var date: String = "default date",
        @XmlAttribute var doneBy: String = "default done by",
        @XmlAttribute var location: String = "default location",
        @field:XmlElement(name="Item")
        @XmlElement var Item: MutableCollection<EventItem> = mutableListOf()
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/Event.cs
using System.Collections.ObjectModel;
using System.Diagnostics.CodeAnalysis;
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    [XmlType("Events")]
    [SuppressMessage("Microsoft.Naming", "CA1716",
        Justification = "Deliberately matching the name specified by the GPSS")]
    public class Event
    {
        [XmlAttribute("date")] public string? Date { get; set; }

        [XmlAttribute("doneBy")] public string? DoneBy { get; set; }

        [XmlAttribute("location")] public string? Location { get; set; }

        [XmlElement("Item")]
        public Collection<RequestPatientRecordItem>? Items { get; set; }
    }
}